src/dlgProfilePreferences.cpp

```diff
@@ -837,65 +837,67 @@ void dlgProfilePreferences::initWithHost(Host* pHost)
 #if !defined(QT_NO_SSL)
     if (QSslSocket::supportsSsl() && pHost->mSslTsl) {
         QSslCertificate cert = pHost->mTelnet.getPeerCertificate();
-        ssl_issuer_label->setText(cert.issuerInfo(QSslCertificate::CommonName).join(","));
-        ssl_issued_label->setText(cert.subjectInfo(QSslCertificate::CommonName).join(","));
-        ssl_expires_label->setText(cert.expiryDate().toString(Qt::LocalDate));
-        ssl_serial_label->setText(QString::fromStdString(cert.serialNumber().toStdString()));
-        checkBox_self_signed->setStyleSheet("");
-        checkBox_expired->setStyleSheet("");
-        ssl_issuer_label->setStyleSheet("");
-        ssl_expires_label->setStyleSheet("");
-        checkBox_ssl->setStyleSheet("");
-
-        if (!pHost->mTelnet.getSslErrors().empty()) {
-            // handle ssl errors
-            notificationAreaIconLabelWarning->show();
-            notificationArea->show();
-            notificationAreaMessageBox->show();
-            //notificationAreaMessageBox->setText(pHost->mTelnet.errorString());
-
-            QList<QSslError> sslErrors = pHost->mTelnet.getSslErrors();
-
-            for (int a = 0; a < sslErrors.count(); a++) {
-                QString thisError = QStringLiteral("<li>%1</li>").arg(sslErrors.at(a).errorString());
-                notificationAreaMessageBox->setText(QStringLiteral("%1\n%2").arg(notificationAreaMessageBox->text(), thisError));
-
-                if (sslErrors.at(a).error() == QSslError::SelfSignedCertificate) {
-                    checkBox_self_signed->setStyleSheet(QStringLiteral("font-weight: bold; background: yellow"));
-                    ssl_issuer_label->setStyleSheet(QStringLiteral("font-weight: bold; color: red; background: yellow"));
-                }
-                if (sslErrors.at(a).error() == QSslError::CertificateExpired) {
-                    checkBox_expired->setStyleSheet(QStringLiteral("font-weight: bold; background: yellow"));
-                    ssl_expires_label->setStyleSheet(QStringLiteral("font-weight: bold; color: red; background: yellow"));
+        if (cert.isNull()) {
+            groupBox_ssl_certificate->hide();
+        } else {
+            ssl_issuer_label->setText(cert.issuerInfo(QSslCertificate::CommonName).join(","));
+            ssl_issued_label->setText(cert.subjectInfo(QSslCertificate::CommonName).join(","));
+            ssl_expires_label->setText(cert.expiryDate().toString(Qt::LocalDate));
+            ssl_serial_label->setText(QString::fromStdString(cert.serialNumber().toStdString()));
+            checkBox_self_signed->setStyleSheet("");
```

Comment:
:bulb: I know it is a little beyond the intended scope of this PR but we should be using `QString()` in these places - not `""` ...